noman.el

```diff
@@ -69,141 +70,115 @@
   (let ((subcommand (button-label button)))
     (noman (format "%s %s" noman--last-command subcommand))))
 
-(defun noman--exec (cmd suffix buffer)
-  "Execute CMD with the help SUFFIX and place the results in BUFFER."
-  (call-process "sh" nil buffer nil "-c" (concat cmd " " suffix)))
-
-(defun noman--make-aws-button (line)
-  "Parse the string LINE for an aws-style command."
-  (let ((first-match
-	 (string-match
-	  "^ +o +\\([A-Za-z0-9\\-]+\\)$"
-	  line)))
-    (when first-match
-      (let ((beg (match-beginning 1)) (end (match-end 1)))
-	(make-button
-	 (+ (line-beginning-position) beg)
-	 (+ (line-beginning-position) end)
-	 'action #'noman--follow-link)))))
-
-(defvar noman-parsing-functions
-  '(("aws" . noman--make-aws-button))
-  "Custom parsing functions to use for specific commands.
-Each function should take a single string containing a line of text,
-and return a button, or nil.")
-
-(defun noman--make-default-button (line)
-  "Parse the string LINE for a default command."
-  (when (string-prefix-p "  " line)
-    (let ((first-match
-	   (string-match
-	    "^ +\\([A-Za-z]+[A-Za-z0-9\\-]+\\):* \\{2\\}.*$"
-	    line)))
-      (when first-match
-	(let ((beg (match-beginning 1)) (end (match-end 1)))
-	  (make-button
-	   (+ (line-beginning-position) beg)
-	   (+ (line-beginning-position) end)
-	   'action #'noman--follow-link))))))
-
-(defun noman--get-button-func (cmd)
```

Comment:
It's often clearer to remove verbs like "make" or "get" from function names and orient them around their return values. e.g. `noman--button-func` instead of `noman--get-button-func`.